includes/class.llms.person.handler.php

```diff
@@ -759,6 +764,15 @@ public static function register( $data = array(), $screen = 'registration', $sig
 			// Signon.
 			if ( $signon ) {
 				llms_set_person_auth_cookie( $person_id, false );
+				LLMS()->events()->record(
+					array(
+						'actor_id'     => $person_id,
+						'object_type'  => 'user',
+						'object_id'    => $person_id,
+						'event_type'   => 'account',
+						'event_action' => 'signon',
+					)
+				);
```

Comment:
I don't like this.

I know this fixes the issue but on reviewing the issue in #1359 it looks to me like there's actually no real reason for `llms_set_person_auth_cookie()` to exist and we could be more generally compatible with the WP core (and other plugins) by rewriting this as:

```suggestion
			// Signon.
			if ( $signon ) {
				wp_signon(
					array(
						'user_login'    => $data['user_login'],
						'user_password' => $data['password'],
					),
					is_ssl()
				);
			}
```

Since we set the `llms_last_login` usermeta data in the `llms_set_person_auth_cookie()` method we lose this data in this change.

However, in looking into this I found that we actually don't set this last login during an actual login, only during a registration.

So we could add a simple function (attached to the `wp_login` action) that will record this metadata and then we will regain this during registration and also add it during an actual login (via WP or LifterLMS or any other plugin that may take over login, say WooCommerce which we actually have an issue with here in that this metadata would never be recorded when using LifterLMS with this plugin). 

We can deprecate a function this way. What do you think?

I ran a quick test of this locally and I don't see any real issues with this change.

Reply:
@thomasplevy agreed.
I honestly tend to be conservative, even if I thought to not call `llms_set_person_auth_cookie()` too.
About the `llms_last_login` meta mmm I would just get rid of its setting. Where do we use it? Since we only set it on registration it's not even a last login, but a first login... 
Any statistic based on it would be fake, I think we use it only as fall-back in the last-seen column (which leverages the account.* events)...

Reply:
@eri-trabiccolo 

> About the llms_last_login meta mmm I would just get rid of its setting. Where do we use it? Since we only set it on registration it's not even a last login, but a first login...
Any statistic based on it would be fake, I think we use it only as fall-back in the last-seen column (which leverages the account.* events)...

I think that at some point it wasn't fake and got lost in an unrelated fix or refactor at some point.

But yes, I think you're right we can probably drop it all together.

Reply:
@thomasplevy 
The `last_login` is used in two places:
- users table
- as fall back for "last seen" in the report's students table 

I think we want to keep this information separated from the "last seen", and that we want to avoid "calculating" it each time. Also, the `last_login` being a user meta, when populating the users table, is retrieved along with the other user metas and cache by wp in a single query.
So I think I'm going to follow your first "advice": attach the usermeta recording (with the current value `current_time('mysql')`) to the `wp_login` action, although I'm not exactly sure where to put this :D